src/main/java/org/wildfly/plugin/tools/client/ToolsModelControllerClientConfiguration.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.plugin.tools.client;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+
+import javax.net.ssl.SSLContext;
+import javax.security.auth.callback.CallbackHandler;
+
+import org.jboss.as.controller.client.ModelControllerClientConfiguration;
+import org.wildfly.plugin.tools.security.ClientCallbackHandler;
+import org.wildfly.security.SecurityFactory;
+
+/**
+ * A configuration used to connect a {@link org.jboss.as.controller.client.ModelControllerClient} or used to connect a
+ * CLI {@code CommandContext}
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+public class ToolsModelControllerClientConfiguration implements ModelControllerClientConfiguration {
+
+    private final ModelControllerClientConfiguration delegate;
+    private final String username;
+    private final String password;
+    private final CallbackHandler callbackHandler;
+
+    public ToolsModelControllerClientConfiguration(final ModelControllerClientConfiguration delegate, final String username,
+            final String password) {
+        this.delegate = delegate;
+        this.username = username;
+        this.password = password;
+        if (delegate.getAuthenticationConfigUri() == null) {
+            callbackHandler = new ClientCallbackHandler(username, password);
+        } else {
+            callbackHandler = delegate.getCallbackHandler();
+        }
+    }
```

Comment:
Out of curiosity, where is the plan to use this? We need to think about how the username and password are stored. Especially since there are public getters for them.

Reply:
Replacement of:
https://github.com/wildfly/wildfly-maven-plugin/blob/main/plugin/src/main/java/org/wildfly/plugin/common/ClientCallbackHandler.java
and
https://github.com/wildfly/wildfly-maven-plugin/blob/main/plugin/src/main/java/org/wildfly/plugin/common/MavenModelControllerClientConfiguration.java